ethcore/src/state/mod.rs

```diff
@@ -310,6 +311,11 @@ impl<B: Backend> State<B> {
 		Ok(state)
 	}
 
+	/// Get a VM factory that can execute on this state.
+	pub fn vm_factory(&self) -> EvmFactory {
```

Comment:
Yep, it's mostly there because of "State::apply" which should really be a `fn apply(&mut State, VMFactory)` in some other module.